pipelines/run-pipelines-on-kubeflow-pipelines/README.md

```diff
@@ -0,0 +1,302 @@
+
+      ![Use labels to produce unique node names](doc/images/label-example.png)
+
+   - `URL`: This is a required input of the '`Download File`' component:
+      ```
+      inputs:
+      - {name: URL, type: String, optional: false, description: 'File URL'}
+      ```
+      The pipeline editor renders component inputs using an editable widget, such as a text box, and, if one was provided, displays input's description. Since this property is marked in the specification as required, the pipeline editor enforces the constraint.
+
+      ![Rendering of a component input in the pipeline editor](doc/images/rendering-of-a-component-input.png) 
+   - `downloaded file`: This is an output of the '`Download File`' component:
+      ```
+      outputs:
+      - {name: downloaded file, type: String, description: 'Content of the downloaded file.'}
+      ```
+      The pipeline editor renders outputs using read-only widgets.
+
+      ![Rendering of a component output in the pipeline editor](doc/images/rendering-of-a-component-output.png)
+
+   - `Component source`: A read-only property that identifies the location from where the component specification was loaded. This property is displayed for informational purposes only.   
+
+1. Enter `https://raw.githubusercontent.com/elyra-ai/examples/master/pipelines/run-pipelines-on-kubeflow-pipelines/data/data.csv` as value for the `URL` input property.
+
+   ![Configure download node](doc/images/configure-download-node.png)
+
+1. Open the properties of the '`Count Rows`' node. The [specification for the underlying component looks as follows](https://raw.githubusercontent.com/elyra-ai/examples/master/pipelines/run-pipelines-on-kubeflow-pipelines/components/count-rows.yaml):
+   ```
+   name: Count Rows
+   description: Count the number of rows in the input file
+
+   inputs:
+   - {name: input file, type: String, optional: false, description: 'Row-based file to be analyzed'}
+
+   outputs:
+   - {name: row count, type: String, description: 'Number of rows in the input file.'}
+   ...
+   implementation:
+   ...
+    command: [
+      python3, 
+      /pipelines/component/src/count-rows.py,
+      --input-file-path,
+      {inputPath: input file},
+   ...
+   ```
+
+   The component requires one input ('`input file`') and produces one output ('`row count`'), which is the number of rows in this file.
+
+   Note that Kubeflow Pipelines passes the input the the implementing Python script as a file handle:
+   ```
+   --input-file-path,
+      {inputPath: input file},
+   ```   
+
+   The pipeline editor takes this as a cue and renders a selector widget for this input:
+
+   ![Configure Count Rows node](doc/images/configure-count-rows-node.png)
+
+   Since the '`Count Rows`' node is only connected to one upstream node ('`Download File`'), you can only choose from the outputs of that node. (An upstream node is a node that is connected to the node and executed before the node.) 
```

Comment:
Another minor suggestion for clarity:
```suggestion
   Since the '`Count Rows`' node is only connected to one upstream node ('`Download File`'), you can only choose from the outputs of that node. (An upstream node is a node that is connected to and executed before the node in question.) 
```